### ISSUE TITLE
`systemd-journald` outputting to `/` (instead of `/var`) partition

### ISSUE THREAD

**Issue by nazarewk** [systemd/systemd#4494]
### Submission type

Bug report
### systemd version the issue has been seen with

> systemd 231
### Used distribution

Arch Linux

> Linux 4.8.2-1-ARCH #1 SMP PREEMPT Mon Oct 17 08:11:46 CEST 2016 x86_64 GNU/Linux
### Expected behaviour you didn't see

> Waiting for mounting of `/var` before `systemd-journald` starts
### Unexpected behaviour you saw

> `systemd-journald` outputs stuff to root partition's `/var/log/journal` instead of `/var`'s `/log/journal`, only user services are logged to proper location.
### Steps to reproduce the problem

I am not sure how to reproduce, my setup is:
- encrypted /boot
- [LVM on LUKS](https://wiki.archlinux.org/index.php/Dm-crypt/Encrypting_an_entire_system#LVM_on_LUKS)
- `/var` has it's own LVM partition
- [fstab.txt](https://github.com/systemd/systemd/files/553756/fstab.txt)
- [services.txt](https://github.com/systemd/systemd/files/553749/services.txt) i am pretty sure i added that RequiresMountFor some time ago and it worked for a while, until October 10th
### After `mount --bind / /mnt`

```
root@dc-krna ..urnal/6cf5f7fc557a4bb69efd0d795d090b08 # journalctl -b -u systemd-journal-flush -u var.mount -u systemd-journald -D /mnt/var/log/journal
-- Logs begin at Mon 2016-10-10 10:37:31 CEST, end at Wed 2016-10-26 17:24:11 CEST. --
Oct 25 15:59:41 dc-krna systemd-journald[315]: Runtime journal (/run/log/journal/) is 8.0M, max 394.6M, 386.6M free.
Oct 25 15:59:42 dc-krna systemd-journald[315]: System journal (/var/log/journal/) is 192.0M, max 3.1G, 2.9G free.
Oct 25 15:59:42 dc-krna systemd-journald[315]: Time spent on flushing to /var is 1.863ms for 2 entries.
Oct 25 15:59:42 dc-krna systemd-journald[315]: Journal started
Oct 25 15:59:44 dc-krna systemd[1]: var.mount: Directory /var to mount over is not empty, mounting anyway.
Oct 25 15:59:44 dc-krna systemd[1]: Mounting /var...
Oct 25 15:59:44 dc-krna systemd[1]: Mounted /var.
Oct 25 15:59:44 dc-krna systemd[1]: Starting Flush Journal to Persistent Storage...
Oct 25 15:59:44 dc-krna systemd[1]: Started Flush Journal to Persistent Storage.
```

**Comment by keszybz** [comment]
Journald tries to be smart and flushes to /var on startup if the directories are present. Does the problem go away if you remove the directory tree underneath the mount point?

**Comment by poettering** [comment]
> Journald tries to be smart and flushes to /var on startup if the directories are present.

We only do that when Storage=auto is set, actually.

**Comment by nazarewk** [comment]
@keszybz this time (I am pretty sure it helped the first time) problem did not go away when i did `rm -Rf /var/log` on offline system (booted from usb stick)

**Comment by poettering** [comment]
This should be fixed by f78273c8dacf678cc8fd7387f678e6344a99405c (part of v233). Closing.
